server/header_context.go

```diff
@@ -104,15 +106,19 @@ func apigeeDynamicDataHeaders(org, env, api string, apiSpec *config.APISpec, fau
 	}
 
 	// Include fault related headers.
-	if fault {
+	if adapterFault != nil && adapterFault.RpcCode == rpc.INTERNAL {
```

Comment:
Hrm, you know you can define a method on `AdapterFault` here that can respond to `nil`? Something like:

```
func (af *AdapterFault) GetRpcCode() rpc.Code {
  if af == nil {
    return rpc.OK
  }

  return af.RpcCode
}
```

(Not saying you have to, but you can simplify these methods a bit if you like)

Reply:
Just wanted to point out it's probably better to do `RpcCode()` since `GetXYZ` is not recommended. I know there are bad examples in this repo as well as the proto-generated pkgs though.

Reply:
Oh gosh, yes! Thanks, I forgot that `GetFoo()` is an antipattern in Go.

Reply:
Thanks for the suggestions!
I think method and field with same name cannot co-exist. Skipping this for now.